hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -1462,8 +1473,14 @@ protected double getCostFromRl(BalancerRegionLoad rl) {
     }
 
     @Override
-    protected double getCostFromRl(BalancerRegionLoad rl) {
-      return rl.getStorefileSizeMB();
+    protected double getCostFromRl(BalancerRegionLoad rl, boolean isPrimaryRegion) {
+      // Do not count replica region's file size, as replica regions serve very little
+      // read requests, this may be changed if there are enough data from production showing
```

Comment:
I don't understand this comment "as replica regions serve very little read requests." All a replica does is serve read requests! Maybe you're trying to say "a replica region by definition contributes nothing to store file load on the region that hosts it."

To this, I think it depends on how this value is being considered. This store file size has a real impact on IO channels and BlockCache; just happenstance impact on local storage IO channels. Notably, compactions impose no load for a replica region. I think it makes sense to continue to consider the store file size of read replicas, to maybe at a reduced weight. Omitting them entirely only makes sense if we can somehow assume that all region servers will serve an equal load of replicas, serving equal work, which is not realistic.

Reply:
As I wrote in the comments, all these factors really impacts system performance.  From one of the production clusters' stats, < 0.01% of requests goes to replica regions, which means most of regions are cold at Region servers. That is the reason I want to remove this factors from balancer. Agreed with you that things could be different with others, make it configurable makes more sense. If it is ok with you, I want to drop this change from this patch and creates a separate issue to track it, probably with a test case as well.